pkg/reconciler/pipelinerun/resources/pipelineref.go

```diff
@@ -41,7 +41,8 @@ import (
 // GetPipelineFunc is a factory function that will use the given PipelineRef to return a valid GetPipeline function that
 // looks up the pipeline. It uses as context a k8s client, tekton client, namespace, and service account name to return
 // the pipeline. It knows whether it needs to look in the cluster or in a remote location to fetch the reference.
-func GetPipelineFunc(ctx context.Context, k8s kubernetes.Interface, tekton clientset.Interface, requester remoteresource.Requester, pipelineRun *v1beta1.PipelineRun) rprp.GetPipeline {
+// OCI bundle and remote resolution pipelines will be verified by trusted resources if the feature is enabled
+func GetPipelineFunc(ctx context.Context, k8s kubernetes.Interface, tekton clientset.Interface, requester remoteresource.Requester, pipelineRun *v1beta1.PipelineRun, verificationPolicies []*v1alpha1.VerificationPolicy) rprp.GetPipeline {
```

Comment:
Why don't we verify the resource in the default case where in-cluster resources are resolved by the local cluster resolver?

Prior to the move, that case seems to be included for verification but after the move the case seems ignored.

Same for task reconciler.

Reply:
Do you mean that the in-cluster resources could be mutated by the mutation webhook when they are applied to the cluster, which might cause the verification fail?

If so, the remote resolver is also able to resolve the in-cluster resources via [cluster resolver](https://github.com/tektoncd/pipeline/blob/main/docs/cluster-resolver.md). Will there be same issue and how are we planning to handle that?

Reply:
I think currently we could just document it, then when we finish high priority features(v1 support, condition update), we can start planning solutions for local resources

Reply:
yes, that's still in the scope, but just to the priority at this moment.